.tools/readmes/runner.py

```diff
@@ -35,61 +44,70 @@ def prepare_scanner(doc_gen: DocGen) -> Optional[Scanner]:
-        help=f"Save a copy of the original README as the 'saved_readme' value specified in config.py ({config.saved_readme}).",
-    )
-    parser.add_argument(
-        "--verbose",
-        action="store_true",
-        help="When set, output verbose debugging info.",
-    )
-    parser.add_argument(
-        "--dry-run",
-        action="store_true",
-        dest="dry_run",
-        help="In dry run, compare current vs generated and exit with failure if they do not match.",
-        default=False,  # Change this to default false when we're ready to use this generally.
-    )
-    parser.add_argument("--no-dry-run", dest="dry_run", action="store_false")
-    parser.add_argument("--check", dest="dry_run", action="store_true")
-    parser.add_argument("--diff", action="store_true", default=False)
-    args = parser.parse_args()
-
-    if "all" in args.languages:
-        args.languages = [*languages]
-
-    if "all" in args.services:
-        args.services = [*doc_gen.services.keys()]
-
-    if args.verbose:
+# Load all examples immediately for cross references. Trades correctness for speed.
+doc_gen = DocGen.from_root(Path(__file__).parent.parent.parent, incremental=True)
+
+
+Language = Enum(
+    "Language", {lang: lang for lang in ([*doc_gen.languages()] + ["all"])}
+)  # type: ignore
+Service = Enum(
+    "Service", {serv: serv for serv in ([*doc_gen.services.keys()] + ["all"])}
+)  # type: ignore
+
+
+def writeme(
+    languages: Annotated[
+        list[Language],  # type: ignore
+        typer.Option(
+            "--language",
+            help="The languages of the SDK.",
+        ),
+    ] = [
+        Language.all  # type: ignore
+    ],  # type: ignore
+    services: Annotated[
+        list[Service],  # type: ignore
+        typer.Option(
+            "--service",
+            help="The targeted service.",
+        ),
+    ] = [
+        Service.all  # type: ignore
+    ],  # type: ignore
+    safe: Annotated[
+        bool,
+        typer.Option(
+            help="Save a copy of the original README as the 'saved_readme' value specified in config.py ({config.saved_readme})."
+        ),
+    ] = True,
+    verbose: Annotated[
+        bool, typer.Option(help="When set, output verbose debugging info.")
+    ] = False,
+    dry_run: Annotated[
+        bool,
+        typer.Option(
+            help="In dry run, compare current vs generated and exit with failure if they do not match."
+        ),
+    ] = False,
+    check: Annotated[bool, typer.Option(help="Alias for --dry-run.")] = False,
+    diff: Annotated[
+        bool, typer.Option(help="Show a diff of READMEs that have changed.")
+    ] = False,
+):
+    if "all" in languages:
+        languages = list(Language)  # type: ignore
+
+    if "all" in services:
+        services = list(Service)  # type: ignore
+
+    if verbose:
         logging.basicConfig(level=logging.DEBUG)
 
-    logging.debug(f"Args configuration: {args}")
+    if check:
+        dry_run = check
```

Comment:
If we're setting dry_run equal to check, why not just use check directly? Or rename the flag `dry_run`?